packages/nextjs/src/editing/utils.ts

```diff
@@ -0,0 +1,302 @@
+import {
+  DesignLibraryRenderPreviewData,
+  EDITING_ALLOWED_ORIGINS,
+  EditingRenderQueryParams,
+  isDesignLibraryMode,
+  PREVIEW_KEY,
+  QUERY_PARAM_EDITING_SECRET,
+} from '@sitecore-content-sdk/core/editing';
+import { DEFAULT_VARIANT } from '@sitecore-content-sdk/core/personalize';
+import { SITE_KEY } from '@sitecore-content-sdk/core/site';
+import { NextApiRequest } from 'next';
+import { NextRequest } from 'next/server';
+import {
+  EDITING_PASS_THROUGH_HEADERS,
+  QUERY_PARAM_VERCEL_PROTECTION_BYPASS,
+  QUERY_PARAM_VERCEL_SET_BYPASS_COOKIE,
+} from './constants';
+import { IncomingHttpHeaders } from 'http';
+import { SERVER_PROPS_ID, STATIC_PROPS_ID } from 'next/constants';
+import { NativeDataFetcher } from '@sitecore-content-sdk/core';
+import { getAllowedOriginsFromEnv } from '@sitecore-content-sdk/core/utils';
+
+/**
+ * Gets editing secret value from request
+ * @param {NextApiRequest | NextRequest} req incoming request
+ * @returns {string | undefined} editing secret value if present
+ */
+export const getEditingSecretFromRequest = (req: NextApiRequest | NextRequest) => {
+  const reqQuery = (req as NextApiRequest).query;
+  const reqUrl = (req as NextRequest).url;
+
+  let secret = undefined;
+
+  if (reqQuery) {
+    // pages router
+    secret = reqQuery[QUERY_PARAM_EDITING_SECRET];
+  } else if (reqUrl) {
+    // app router
+    const url = new URL(reqUrl);
+    secret = url.searchParams.get(QUERY_PARAM_EDITING_SECRET);
+  }
+
+  return secret;
+};
+
+/**
+ * Parses query string and its parameters to required editing parameters
+ * @param {{ [key: string]: string | undefined }} query query string values
+ * @returns {EditingRenderQueryParams} editing parameters
+ */
+export const getEditingParams = (query: {
+  [key: string]: string | undefined;
+}): { [key: string]: string | undefined } => {
+  const params = isDesignLibraryMode(query.mode)
+    ? {
+        itemId: query.sc_itemid,
+        componentUid: query.sc_uid,
+        renderingId: query.sc_renderingId,
+        language: query.sc_lang,
+        site: query.sc_site,
+        mode: query.mode,
+        dataSourceId: query.dataSourceId,
+        version: query.sc_version,
+      }
+    : {
+        site: query.sc_site,
+        itemId: query.sc_itemid,
+        language: query.sc_lang,
+        // for sc_variantId we may employ multiple variants (page-layout + component level)
+        // they will be separated by commas (,)
+        variantIds: query.sc_variant || DEFAULT_VARIANT,
+        version: query.sc_version,
+        mode: query.mode,
+        layoutKind: query.sc_layoutKind,
+      };
+  return params;
+};
+
+/**
+ * Filters out Next.js preview cookies from a cookie string or array
+ * @param {string | string[] | null} cookies cookie header value
+ * @returns {string[] | null} filtered cookies
+ */
+export const getFilteredCookies = (cookies: string | string[] | null) => {
```

Comment:
Method name doesn't say anything about Next or preview, making method purpose unclear, let's name it as `extractNextPreviewCookies`